jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java

```diff
@@ -209,7 +215,27 @@ private static String prefix(String prefix) {
   }
 
   public void run(String peerServiceTag) {
-    run(indexDate("jaeger-span"), indexDate("jaeger-dependencies") ,peerServiceTag);
+
+    String[] readIndices;
+    String[] writeIndex;
+
+    // read spans from alias index name
+    if (this.useReadAlias) {
+      readIndices = new String[]{prefix(indexPrefix) + "jaeger-span-read", prefixBefore19(indexPrefix) + "jaeger-span-read"};
+    }
+    else {
+      readIndices = indexDate("jaeger-span");
+    }
+
+    // write dependencies to alias index name
+    if (this.useWriteAlias) {
+      writeIndex = new String[] {prefix(indexPrefix) + "jaeger-dependencies-write"};
```

Comment:
Heya, just raising a bug I found from this - I enabled readAlias and writeAlias and realised that when it finds no dependency on the first loop, it gives me this error
```
 22/04/13 20:05:46 INFO ElasticsearchDependenciesJob: Running Dependencies job for 2022-04-13T00:00Z, reading from jaeger-span-read index, result storing to jaeger-dependencies-write     │
│ 22/04/13 20:05:56 INFO ElasticsearchDependenciesJob: Done, 0 dependency objects created                                                                                                   │
│ Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
```

It is because `writeIndex` is one element in the array but `readIndices` has two.

Will need to do 
```suggestion
      writeIndex = new String[] {prefix(indexPrefix) + "jaeger-dependencies-write", prefixBefore19(indexPrefix) + "jaeger-dependencies-write"};
```

Reply:
Thanks for your feedback and the suggested change.
I, as you might have guessed, did not use the older prefixBefore19 style so I did not run into this issue.
But fair point. I rebased my PR and added your suggested change.

At some point it might be sensible to drop support for two styles and cleanup the code in this regard.
@albertteoh Do you mind taking another look and maybe provide a path to have this whole PR merged?